snakemake/sourcecache.py

```diff
@@ -148,41 +135,37 @@ def __init__(
         self.repo_path = repo_path
         self.path = path
 
+    def get_basedir(self):
+        return self.__class__(
+            tag=self.tag,
+            commit=self.commit,
+            ref=self._ref,
+            repo_path=self.repo_path,
+            path=os.path.dirname(self.path),
+        )
+
     def get_path_or_uri(self):
-        return "git+file://{}/{}@{}".format(self.repo_path, self.path, self.ref)
+        return "git+{}/{}@{}".format(self.repo_path, self.path, self.ref)
 
     def join(self, path):
         return LocalGitFile(
             self.repo_path,
-            "/".join((self.path, path)),
-            tag=self.tag,
-            ref=self.ref,
-            commit=self.commit,
-        )
-
-    def get_basedir(self):
-        return self.__class__(
-            repo_path=self.repo_path,
-            path=os.path.dirname(self.path),
+            os.path.normpath("/".join((self.path, path))),
             tag=self.tag,
+            ref=self._ref,
             commit=self.commit,
-            ref=self.ref,
         )
 
     def is_persistently_cacheable(self):
         return False
 
     def get_filename(self):
-        return posixpath.basename(self.path)
+        return os.path.basename(self.path)
```

Comment:
What is the reason to not use posixpath here?

Reply:
old habit from my side! Are there any benefits of using posixpath compared to os.path?

Reply:
It always uses /, whereas path uses \ on windows (which is not what we want for urls).